admin/src/dataProvider.tsx

```diff
@@ -5,25 +5,37 @@ const apiUrl = process.env.REACT_APP_API_URL;
 const httpClient = fetchUtils.fetchJson;
 
 export const dataProvider: DataProvider = {
-  getList: (resource, params) => {
+
+  getList: async (resource, params) => {
     const { page, perPage } = params.pagination;
     const { field, order } = params.sort;
+
     const query = {
       sort: JSON.stringify([field, order]),
       range: JSON.stringify([(page - 1) * perPage, page * perPage - 1]),
       filter: JSON.stringify(params.filter),
     };
     const url = `${apiUrl}/${resource}?${stringify(query)}`;
 
-    return httpClient(url).then(({ headers, json }) => {
+    return httpClient(url, {
```

Comment:
Rather than filling it in for each function, check this link.  : https://marmelab.com/react-admin/DataProviders.html#adding-custom-headers